pytorch_lightning/callbacks/hpu_stats_monitor.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (C) 2021 Habana Labs, Ltd. an Intel Company
+# All Rights Reserved.
+#
+# Unauthorized copying of this file or any element(s) within it, via any medium
+# is strictly prohibited.
+# This file contains Habana Labs, Ltd. proprietary and confidential information
+# and is subject to the confidentiality and license agreements under which it
+# was provided.
+#
+
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+hpu Stats Monitor
+=================
+
+Monitor and logs hpu stats during training.
+
+"""
+from typing import Any, Dict, List, Optional, Tuple
+
+import torch
+
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks.base import Callback
+from pytorch_lightning.utilities import rank_zero_only
+
+
+class HPUStatsMonitor(Callback):
```

Comment:
This is just a callback for stats being used at present. We have plans to integrate some additional features in this.

Reply:
> We have plans to integrate some additional features in this.

You mean planned additional features don't fit into `HPUAccelerator.get_device_stats()`? I don't see why `HPUStatsMonitor` is being added while the other two `XLAStatsMonitor` and `GPUStatsMonitor` will be removed soon.

Reply:
No. When we need to integrate backend specific API/calls in the device stat monitor, it will pollute the base class having segregation for GPU, HPU etc.